### PR TITLE
arktos network controller manages network object status phase

### PR DESCRIPTION
**Pull request by h-w-chen**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
arktos network controller will manage network object status.phase field, e.g. for mizar typed network, if status.DNSServiceIP is present, phase shall be set to Ready.

Besides Ready phase, this PR also manage Pending and Terminating phases. Howeer, this PR does not put network in Failed or Unknown phase; we need to figure out the proper communication for external provider to notice this arktos contrroller first.

**Special notes for your reviewer**:
network metadata.finalizer is not in this PR. finalizer will consider later as part of network deletion.
 
**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: docs/design-proposals/multi-tenancy/multi-tenancy-network.md [hunk 1]
```diff
@@ -146,6 +146,21 @@ If these settings are set on a pod attached to a flat network, the settings will
 
 (**TBD: do we plan to support multiple network providers? How would it impact the type definition?**)
 
+#### Network Status
+The network resource could have status.phase explicitly indicating its current phase in the whole life cycle. Arktos network controller is responsible to update such field, based on appropriate conditions.
```
Comment (chenqianfzh) [inline docs/design-proposals/multi-tenancy/multi-tenancy-network.md:150]: update such field -> update this field?

File: cmd/arktos-network-controller/app/controller_test.go [hunk 2]
```diff
@@ -197,7 +194,8 @@ func TestManageNonFlatNetwork(t *testing.T) {
 				},
 				Status: corev1.ServiceStatus{},
 			},
-			expectingError: false,
+			expectingError:   false,
+			expectedNetPhase: "Pending",
```
Comment (chenqianfzh) [inline cmd/arktos-network-controller/app/controller_test.go:198]: v1.NetworkingPending?

Please also check the hard-coded net phases and replace them with the const.

**Review by chenqianfzh — APPROVED** [review]
LGTM. some minor comments about coding.

File: docs/design-proposals/multi-tenancy/multi-tenancy-network.md (hunk 1 above)
Comment (h-w-chen) [reply docs/design-proposals/multi-tenancy/multi-tenancy-network.md:150]: will change as suggested.

File: cmd/arktos-network-controller/app/controller_test.go (hunk 2 above)
Comment (h-w-chen) [reply cmd/arktos-network-controller/app/controller_test.go:198]: test code is fine to have string literal.

**Pull request merged**